site/docs/plugins/conversations.md

```diff
@@ -7,1190 +7,1527 @@ next: false
+The plugin has to somehow remember this data, right?
+
+Yes.
+
+### Conversations Store State
+
+Two types of data are being stored in a database.
+By default, it uses a lightweight in-memory database that is based on a `Map`, but you can [use a persistent database](#persisting-conversations) easily.
+
+1. The conversations plugin stores all updates.
+2. The conversations plugin stores all return values of `conversation.external` and the results of all API calls.
 
-In total, your code should now roughly look like this:
+This is not an issue if you only have a few dozen updates in a conversation.
+(Remember that during long polling, every call to `getUpdates` retrieves up to 100 updates, too.)
+
+However, if your conversation never exits, this data will accumulate and slow down your bot.
+**Avoid infinite loops.**
+
+### Conversational Context Objects
+
+When a conversation is executed, it uses the persisted updates to generate new context objects from scratch.
+**These context objects are different from the context object in the surrounding middleware.**
+For TypeScript code, this also means that you now have two [flavors](../guide/context#context-flavors) of context objects.
+
+- **Outside context objects** are the context objects that your bot uses in middleware.
+  They give you access to `ctx.conversation.enter`.
+  For TypeScript, they will at least have `ConversationFlavor` installed.
+  Outside context objects will also have other properties defined by plugins that you installed via `bot.use`.
+- **Inside context objects** (also called **conversational context objects**) are the context objects created by the conversations plugin.
+  They can never have access to `ctx.conversation.enter`, and by default, they also don't have access to any plugins.
+  If you want to have custom properties on inside context objects, [scroll down](#using-plugins-inside-conversations).
+
+You have to pass both the outside and the inside context types to the conversation.
+The TypeScript setup therefore typically looks as follows:
 
 ::: code-group
 
-```ts [TypeScript]
-import { Bot, Context, session } from "grammy";
+```ts [Node.js]
+import { Bot, type Context } from "grammy";
 import {
   type Conversation,
   type ConversationFlavor,
-  conversations,
-  createConversation,
 } from "@grammyjs/conversations";
 
-type MyContext = Context & ConversationFlavor;
-type MyConversation = Conversation<MyContext>;
+// Outside context objects (knows all middleware plugins)
+type MyContext = ConversationFlavor<Context>;
+// Inside context objects (knows all conversation plugins)
+type MyConversationContext = Context;
 
+// Use the outside context type for your bot.
 const bot = new Bot<MyContext>("");
 
-bot.use(session({ initial: () => ({}) }));
-bot.use(conversations());
+// Use both the outside and the inside type for your conversation.
+type MyConversation = Conversation<MyContext, MyConversationContext>;
+
+// Define your conversation and make sure
```

Comment:
> // Define your conversation and make sure

Umm... make sure of what?

Reply:
Read the following comments, I think.

Reply:
But the following comments are just information.
 ```ts
  // All context objects inside the conversation are
  // of type `MyConversationContext`.
  const ctx1 = await conversation.wait();

  // The outside context object can be accessed
  // via `conversation.external` and it is inferred to be
  // of type `MyContext`.
  const session = await conversation.external((ctx) => ctx.session);
```

Reply:
Yes.
`... and make sure all context objects ...`
`... and make sure the outside context object ...`

Reply:
I don't remember what I meant here but I think it's just wrong. If I intended to write this that way, then it's still confusing and should be improved.
```suggestion
// Define your conversation.
```
I'd just do this for now. If we have a better idea or want to convey more information in the future, we can fix things later.

Can you adjust your translations if we fix this now? If yes, feel free to commit my suggestion right away.

Reply:
Even when the both comments are put together, they still don't make sense. So, I shortened the comment to just "Define your conversation" in the ID translation